src/storage-preview/azext_storage_preview/tests/latest/test_storage_account_scenarios.py

```diff
@@ -2,156 +2,121 @@
+from ...profiles import CUSTOM_MGMT_PREVIEW_STORAGE
 from azure_devtools.scenario_tests import AllowLargeResponse
 
 
-@api_version_constraint(CUSTOM_MGMT_STORAGE, min_api='2016-12-01')
-class StorageAccountTests(StorageScenarioMixin, ScenarioTest):
-    @api_version_constraint(CUSTOM_MGMT_STORAGE, min_api='2017-06-01')
-    @ResourceGroupPreparer(location='southcentralus')
-    def test_create_storage_account_with_assigned_identity(self, resource_group):
-        name = self.create_random_name(prefix='cli', length=24)
-        cmd = 'az storage account create -n {} -g {} --sku Standard_LRS --assign-identity --https-only'.format(name, resource_group)
-        result = self.cmd(cmd).get_output_in_json()
-
-        self.assertIn('identity', result)
-        self.assertTrue(result['identity']['principalId'])
-        self.assertTrue(result['identity']['tenantId'])
-
-    @api_version_constraint(CUSTOM_MGMT_STORAGE, min_api='2017-06-01')
-    @ResourceGroupPreparer(location='southcentralus')
-    def test_update_storage_account_with_assigned_identity(self, resource_group):
-        name = self.create_random_name(prefix='cli', length=24)
-        create_cmd = 'az storage account create -n {} -g {} --sku Standard_LRS --https-only '.format(name, resource_group)
-        self.cmd(create_cmd, checks=[JMESPathCheck('identity', None)])
-
-        update_cmd = 'az storage account update -n {} -g {} --assign-identity'.format(name, resource_group)
-        result = self.cmd(update_cmd).get_output_in_json()
-
-        self.assertIn('identity', result)
-        self.assertTrue(result['identity']['principalId'])
-        self.assertTrue(result['identity']['tenantId'])
-
-    @AllowLargeResponse()
-    @ResourceGroupPreparer(parameter_name_for_location='location')
-    def test_create_storage_account(self, resource_group, location):
-        name = self.create_random_name(prefix='cli', length=24)
-
-        self.cmd('az storage account create -n {} -g {} --sku {} -l {} --https-only'.format(
-            name, resource_group, 'Standard_LRS', location))
-
-        self.cmd('storage account check-name --name {}'.format(name), checks=[
-            JMESPathCheck('nameAvailable', False),
-            JMESPathCheck('reason', 'AlreadyExists')
+@api_version_constraint(CUSTOM_MGMT_PREVIEW_STORAGE, min_api='2016-12-01')
+class StorageAccountNetworkRuleTests(StorageScenarioMixin, ScenarioTest):
+    @api_version_constraint(CUSTOM_MGMT_PREVIEW_STORAGE, min_api='2017-06-01')
+    @ResourceGroupPreparer(name_prefix='cli_test_storage_service_endpoints')
+    @StorageAccountPreparer()
+    def test_storage_account_network_rules(self, resource_group):
+        kwargs = {
+            'rg': resource_group,
+            'acc': self.create_random_name(prefix='cli', length=24),
+            'vnet': 'vnet1',
+            'subnet': 'subnet1'
+        }
+        self.cmd('storage account create -g {rg} -n {acc} --bypass Metrics --default-action Deny --https-only'.format(**kwargs),
```

Comment:
Since we create storage account using command, is `@StorageAccountPreparer()` needed?

Reply:
storage account preparer is only used to create storage account with basic properties. Here it is to create storage account with some advanced network features.